### PR TITLE
Janitorial work

### PR DESCRIPTION
**Pull request by andlaus**
This PR contains an assortment of smaller fixes, in particular it introduces a "non-strict" mode for `list` and `compare` and cleans up various minor code quality issues in the `compare` tool.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/cli/compare.py [hunk 1]
```diff
@@ -558,6 +560,14 @@ def add_subparser(subparsers: "argparse._SubParsersAction") -> None:
     )
     _parser_utils.add_pdx_argument(parser)
 
+    parser.add_argument(
+        "--no-strict",
+        action="store_true",
+        default=False,
+        required=False,
+        help="Load the dataset in non-strict mode (which is more robust might lead to undefined behavior)",
```
Comment (kayoub5) [inline odxtools/cli/compare.py:568]: nitpick: missing "but" between "robust" and "might"

File: odxtools/cli/compare.py [hunk 2]
```diff
@@ -599,6 +609,9 @@ def run(args: argparse.Namespace) -> None:
     task = Comparison()
     task.param_detailed = args.no_details
 
+    if args.no_strict:
+        odxtools.exceptions.strict_mode = False
```
Comment (kayoub5) [inline odxtools/cli/compare.py:613]: nitpick: `odxtools.exceptions.strict_mode = not args.no_strict`

File: odxtools/cli/compare.py [hunk 3]
```diff
@@ -718,3 +731,5 @@ def run(args: argparse.Namespace) -> None:
     else:
         # no databases & no variants specified
         print("Please specify either a database or variant for a comparison")
+
+    odxtools.exceptions.strict_mode = True
```
Comment (kayoub5) [inline odxtools/cli/compare.py:735]: suggestion: read value then restore it, more fool proof

File: odxtools/cli/list.py [hunk 4]
```diff
@@ -130,6 +132,14 @@ def add_subparser(subparsers: "argparse._SubParsersAction") -> None:
     )
     _parser_utils.add_pdx_argument(parser)
 
+    parser.add_argument(
```
Comment (kayoub5) [inline odxtools/cli/list.py:135]: Add as a global argument, instead of copy pasting in each sub command

File: odxtools/cli/list.py [hunk 5]
```diff
@@ -130,6 +132,14 @@ def add_subparser(subparsers: "argparse._SubParsersAction") -> None:
...
+        "--no-strict",
+        action="store_true",
```
Comment (kayoub5) [inline odxtools/cli/list.py:137]: suggestion: use 
BooleanOptionalAction
see https://stackoverflow.com/a/68837302/1438522

**Review by kayoub5 — CHANGES_REQUESTED** [review]
First three commits look ok and could be merged, last one is more of a feature that merits its own PR

File: odxtools/cli/list.py (hunk 5 above)
Comment (andlaus) [reply odxtools/cli/list.py]: as far as I can see this was added in python 3.9, i.e. python 3.8 does not support this? (luckily we're slowly closing in on the EOL of Ubuntu 20.04, when we can raise the minimum python version to 3.10...)

**Pull request merged**